src/main/java/com/adroll/cantor/HLLByteArray.java

```diff
@@ -14,9 +14,16 @@
 
     private byte[][] b_data;
 
-    public HLLByteArray(int length) {
+    public HLLByteArray(int length) throws Exception {
+        if (Integer.bitCount(length) != 1) {
```

Comment:
We may want to make this `if Integer.bitCount(length) !=1 || length <0` since bitcount(1<<31) ==1